project/src/main/java/com/google/sps/servlets/QueryServlet.java

```diff
@@ -48,17 +53,38 @@ void init(PlacesFetcher inputFetcher) {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    ImmutableList<Place> filteredPlaces;
     try {
-      //TODO(M1): add call to filterer
-      response.setContentType("application/json");
-      response.getWriter().write(new Gson().toJson(
-          Places.randomSort(fetcher.fetch())
-          .stream()
-          .limit(MAX_NUM_PLACES_TO_RECOMMEND)
-          .collect(Collectors.toList())
-      ));
+      UserPreferences userPrefs =
+          UserPreferences.builder()
+              .setMinRating(Float.parseFloat(request.getParameter("rating")))
+              .setMaxPriceLevel(Integer.parseInt(request.getParameter("price")))
+              .setOpenNow(Integer.parseInt(request.getParameter("open")) != 0)
+              .setLocation(getLatLngFromString(request.getParameter("location")))
+              .setCuisines(ImmutableList.copyOf(request.getParameter("cuisines").split(",")))
```

Comment:
Redundant space?

Reply:
where?

Reply:
Hm, I guess something was misleading in the GitHub UI. Sorry!